From 4ace7cf1db75577e9c8eeb4826c9a04ac9bef4a9 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Mon, 28 Sep 2020 15:06:14 +0200
Subject: [PATCH] simpler warning to avoid DivisionByZeroError exception in 8

---
 tests/270_imagick_restoreErrorHandler.phpt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/270_imagick_restoreErrorHandler.phpt b/tests/270_imagick_restoreErrorHandler.phpt
index 77eeb17..ba2f5d1 100644
--- a/tests/270_imagick_restoreErrorHandler.phpt
+++ b/tests/270_imagick_restoreErrorHandler.phpt
@@ -16,7 +16,7 @@ catch(ImagickException $ie) {
 }
 
 try {
-	$x = @(5 / 0);
+	$x = @$x;
 	echo "Normal warning is suppressed".PHP_EOL;
 }
 catch(\Exception $e) {
@@ -27,4 +27,4 @@ catch(\Exception $e) {
 ?>
 --EXPECTF--
 Normal exception
-Normal warning is suppressed
\ No newline at end of file
+Normal warning is suppressed
